### PR TITLE
Fix external link regex

### LINKED ISSUE #189: Bug: Removing external links erases everything between
**Versions**
Longform version: 2.0.3
Obsidian version: 1.3.7
OS [e.g. macOS, Windows, iOS, Android]: Windows
Theme: Minimal
Other plugins that you think might be relevant here:

**Describe the bug**
When two external links are in the same paragraph, the text between them, and the second link, is completely removed from the output.

**To Reproduce**
1. Create a new scene with the following content:
```markdown
[[David]] and [[Ashley]] are two wikilinks to characters.

[David](Character/Characters/David.md) and [Ashley](Character/Characters/Ashley.md) are two external links to the same characters.
```
2. Make sure the "Remove Links" step is part of the scene compilation
3. Click Compile

**Expected behavior**
The scene should be compiled to:
```markdown
David and Ashley are two wikilinks to characters.

David and Ashley are two external links to the same characters.
```

**Screenshots**
Capture

**Additional context**

### PR DESCRIPTION
**Pull request by b-camphart**
External link regex was greedily capturing any character except newlines 
This small change makes it capture as few characters as possible

fix #189

### CODE REVIEW

**Comment by b-camphart** [comment]
@kevboh curious if I should tackle #95 here too, since I'm touching the Regex anyway.

**Comment by kevboh** [comment]
> #95

Please do! Sorry for the late reply.

**Comment by b-camphart** [comment]
fix #95

File: test/compile/steps/remove-links.test.ts [hunk 1]
```diff
@@ -9,7 +9,11 @@ describe("Removing Links", () => {
         expect(replaceWikiLinks("[[Filename]]\n[[Other filename]]")).toEqual("Filename\nOther filename")
         expect(replaceWikiLinks("[[Filename|Alias]]")).toEqual("Alias")
         expect(replaceWikiLinks("[[Filename#^blockId|Alias]]")).toEqual("Alias")
+        expect(replaceWikiLinks("[[Filename#^blockId]]")).toEqual("Filename#^blockId")
```
Comment (kevboh) [inline test/compile/steps/remove-links.test.ts:12]: Hmm—I would think that both this and `[[]]` should produce the empty string. I suppose it's kind of undefined...I'm not sure why someone would put a bare link into a manuscript. Okay, let's leave as-is for now.

**Review by kevboh — APPROVED** [review]
Thank you, especially for the tests!

File: test/compile/steps/remove-links.test.ts (hunk 1 above)
Comment (b-camphart) [reply test/compile/steps/remove-links.test.ts:12]: It's a little odd, but I just wanted to cover all the edge cases.  My driving principle here was to have it do exactly what obsidian itself does when it renders these.  If you don't embed a block id reference, it appears like this.  If you just have empty `[[]]` then it just renders them as-is.
